### ISSUE TITLE
Issues with installing caaph via clusterctl 

### ISSUE THREAD

**Issue by nikParasyr** [kubernetes-sigs/cluster-api-addon-provider-helm#131]
Currently when trying to install caaph via clusterctl (1.5.2 version) gives the following:
```
❯ clusterctl init --addon helm                                                                                                                                                                  
Fetching providers
Error: failed to get provider components for the "helm" provider: failed to read "addon-components.yaml" from provider's repository "addon-helm": failed to download files from GitHub release v0.1.0-alpha.3-rc.0: failed to get file "addon-components.yaml" from "v0.1.0-alpha.3-rc.0" release
```

This is wrong in 2 ways:
- the version that is being installed is not the latest available: `v0.1.0-alpha.3-rc.0` vs `v0.1.0-alpha.10`
- The `v0.1.0-alpha.3-rc.0` version cannot be installed because in the release [assets](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/releases/tag/v0.1.0-alpha.3-rc.0)  the components are `add-on-components.yaml` instead of `addon-components.yaml` that clusterctl expects

Defining a version such as `clusterctl init --addon helm:v0.1.0-alpha.10` works just fine

/kind bug

**Comment by PatrickLaabs** [comment]
Any updates on this?

**Comment by PatrickLaabs** [comment]
Mhh.. I am a little bit confused.
Looking at the clusterctl code, I can find the URL that is directed to the addon-components.yaml. But I run into a "Not Found" error message.

Here is the link:
https://github.com/kubernetes-sigs/cluster-api/blob/8bef2f37fa884a30789df0117e448983d877294f/cmd/clusterctl/client/config/providers_client.go#L339C19-L339C123

**Comment by PatrickLaabs** [comment]
Ok, now it makes sense..

clusterctl version: 1.5.3

Running locally `clusterctl init --addon helm`
produces the following error:
```
  ~ clusterctl init --addon helm
Fetching providers
Error: failed to get provider components for the "helm" provider: failed to read "addon-components.yaml" from provider's repository "addon-helm": failed to download files from GitHub release v0.1.0-alpha.3-rc.0: failed to get file "addon-components.yaml" from "v0.1.0-alpha.3-rc.0" release
```

Looking the the releases...

*v0.1.0-alpha.3-rc.0*:
Does NOT have a addon-components.yaml, instead it is add-on-components.yaml.

Can be verified in the url:
https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/releases/download/v0.1.0-alpha.3-rc.0/add-on-components.yaml

*v0.1.0-alpha.10*
Does have the correctly named yaml-file 'addon-components.yaml', which is the one expected from the clusterctl binary. (Seeing my previous comment with the link to the code).

https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/releases/download/v0.1.0-alpha.10/addon-components.yaml

And thats why running 
`clusterctl init --addon helm:v0.1.0-alpha.10` works.

```
➜  ~ clusterctl init --addon helm:v0.1.0-alpha.10
Fetching providers
Installing cert-manager Version="v1.13.1"
Waiting for cert-manager to be available...
Installing Provider="cluster-api" Version="v1.5.3" TargetNamespace="capi-system"
Installing Provider="bootstrap-kubeadm" Version="v1.5.3" TargetNamespace="capi-kubeadm-bootstrap-system"
Installing Provider="control-plane-kubeadm" Version="v1.5.3" TargetNamespace="capi-kubeadm-control-plane-system"
Installing Provider="addon-helm" Version="v0.1.0-alpha.10" TargetNamespace="caaph-system"

Your management cluster has been initialized successfully!

You can now create your first workload cluster by running the following:

  clusterctl generate cluster [name] --kubernetes-version [version] | kubectl apply -f -
```

What makes me wonder, is, that.. clusterctl seem per default to target the 'latest' release on this github page. But why is it even targeting 'v0.1.0-alpha.3-rc.0'?
The 'latest' label is also on the release 'v0.1.0-alpha.10' here on github.

**Comment by nvermande** [comment]
Hi, what is the status on that now?

**Comment by Jont828** [comment]
@PatrickLaabs @nikParasyr Thanks for bringing this up. Yes, you're correct in that in older versions, the components file was named `add-on-components.yaml` but got renamed to `addon-components.yaml` due to feedback from Cluster API maintainers. 

It seems to me like this issue is that it's failing to find the latest release. Off the top of my head, I'm wondering if it's because there's code that is sorting all the releases by alphabetical order and sees `v0.1.0-alpha.1 < v0.1.0-alpha.10 < v0.1.0-alpha.3-rc.0` due to a naive method of parsing.

**Comment by Jont828** [comment]
Also started a [Slack thread](https://kubernetes.slack.com/archives/C8TSNPY4T/p1701302926308429) about this. It might be worth opening an issue in CAPI as well if we need to make a change to the clusterctl code.

**Comment by Jont828** [comment]
@nikParasyr @nvermande It looks like by semver rules, anything release candidate is considered the later than an alpha release. I guess semver doesn't allow us to have release candidates for alpha releases, meaning `v0.1.0-alpha.3-rc.0` is probably an invalid semver. I went ahead and deleted those releases since they don't work with clusterctl anyway due to the naming of `add-on-components.yaml`. I published a `v0.1.1-alpha.0` release, which is identical to `v0.1.0-alpha.10` but will be sorted properly in clusterctl.
